From 1280e457b32cabe8f43c82ea773ff557706affa3 Mon Sep 17 00:00:00 2001
From: Oliver Brown <oliver.brown@nxp.com>
Date: Wed, 26 Sep 2018 18:32:16 -0500
Subject: [PATCH 4772/5242] MLK-19746-2 arm64: dts: imx8mq/qm: Update display
 port configurations

commit  835e2d040a69958e6a5e0e7b875b354922a4841e from
https://source.codeaurora.org/external/imx/linux-imx.git

Added the defaults for iMX8QM and iMX8MQ Display Port.

Signed-off-by: Oliver Brown <oliver.brown@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8mq-ddr3l-arm2.dts   |   17 +++++++++++++++++
 .../arm64/boot/dts/freescale/fsl-imx8mq-evk-dp.dts |    6 +++++-
 .../dts/freescale/fsl-imx8qm-lpddr4-arm2-dp.dts    |    4 ++++
 3 files changed, 26 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-ddr3l-arm2.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-ddr3l-arm2.dts
index 25bbe03..58926e2 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-ddr3l-arm2.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-ddr3l-arm2.dts
@@ -352,3 +352,20 @@
 	vmmc-supply = <&reg_usdhc2_vmmc>;
 	status = "okay";
 };
+
+&dcss {
+	status = "okay";
+
+	disp-dev = "hdmi_disp";
+};
+
+&hdmi {
+	compatible = "fsl,imx8mq-dp";
+
+	dp-lane-mapping = <0x4e>;
+	dp-link-rate = <0x14>;
+	dp-num-lanes = <0x4>;
+
+	status = "okay";
+};
+
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dp.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dp.dts
index 057c0e6..165ba91 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dp.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dp.dts
@@ -16,7 +16,11 @@
 
 &hdmi {
 	compatible = "fsl,imx8mq-dp";
-	lane_mapping = <0xc6>;
+
+	dp-lane-mapping = <0xc6>;
+	dp-link-rate = <0xa>;
+	dp-num-lanes = <0x4>;
+
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-dp.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-dp.dts
index de73e301..33b2cc0 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-dp.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-dp.dts
@@ -48,5 +48,9 @@
 					<&clk IMX8QM_HDMI_AV_PLL_CLK>,
 					<&clk IMX8QM_HDMI_AV_PLL_CLK>;
 
+	dp-lane-mapping = <0x1b>;
+	dp-link-rate = <0xa>;
+	dp-num-lanes = <0x4>;
+
 	status = "okay";
 };
-- 
1.7.9.5

